terminal.go

```diff
@@ -405,6 +405,31 @@ func (g *LightningTerminal) Run(ctx context.Context) error {
 		g.statusMgr.SetErrored(
 			subservers.LIT, "could not start Lit: %v", startErr,
 		)
+
+		// Stop lnd promptly to avoid it continuing to run after a
+		// failed startup (for example, calling into failed
+		// sub-servers).
+		if client, err := g.basicLNDClient(); err == nil {
+			stopCtx, cancel := context.WithTimeout(
+				ctx, 5*time.Second,
+			)
+			defer cancel()
+
+			_, err := client.StopDaemon(
+				stopCtx, &lnrpc.StopRequest{},
+			)
+			if err != nil {
+				log.Warnf("Error stopping lnd after failed "+
+					"start: %v", err)
+			}
+		}
+
+		if err := g.shutdownSubServers(); err != nil {
+			log.Errorf("Error shutting down after failed start: %v",
+				err)
+		}
+
+		return startErr
```

Comment:
Similar to what @ellemouton has already commented, this will now shutdown `litd` before the `shutdownInterceptor.ShutdownChannel()` has been triggered, which we would like to avoid.